### PR TITLE
Move select2 lib to core

### LINKED ISSUE #31338: Send mail notification for shared files breaks if Collaborative Tags-App is disabled
### Steps to reproduce
1. Disable _Collaborative tags_ in the apps section
2. Enable _"Allow users to send mail notification for shared files"_ in sharing settings
3. Create a public link share

### Expected behaviour
Modal should should show sharing-form

### Actual behaviour
Empty modal opens

### Server configuration
**Operating system**:
any 
**Web server:**
any
**Database:**
any
**PHP version:**
any
**ownCloud version:** (see ownCloud admin page)
10.0.8

### Problem

select2 lib is provided by the Collaborative Tags-App and used in the mail notification process.

#### Javascript console log

`this.$el.find(...).select2 is not a function`

### PR DESCRIPTION
**Pull request by lordelix**
## Description
Move the select2 js library from systemtags app to core for global usage.

## Related Issue

Fixes: https://github.com/owncloud/core/issues/31338

## How Has This Been Tested?

Manual tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: apps/systemtags/appinfo/app.php [hunk 1]
```diff
@@ -28,10 +28,6 @@
 $eventDispatcher->addListener(
 	'OCA\Files::loadAdditionalScripts',
 	function () {
-		// FIXME: no public API for these ?
-		\OC_Util::addVendorScript('select2/select2');
-		\OC_Util::addVendorStyle('select2/select2');
```
Comment (patrickjahns) [inline apps/systemtags/appinfo/app.php]: where has this gone?

File: lib/private/legacy/template.php [hunk 2]
```diff
@@ -132,6 +132,7 @@ public static function initTemplateEngine($renderAs) {
 
 			OC_Util::addScript('oc-backbone', null, true);
 			OC_Util::addVendorScript('core', 'backbone/backbone', true);
+			OC_Util::addVendorScript('core', 'select2/select2', true);
```
Comment (PVince81) [inline lib/private/legacy/template.php]: did you miss adding the style as well or did it magically worked for some strange reasons ?

Comment (PVince81) [reply lib/private/legacy/template.php]: seems it's already here: https://github.com/owncloud/core/pull/31498/files#diff-f7a9bb6da381ad3d1651acb5f47162a1R115

please move it down here for better readability

File: apps/systemtags/appinfo/app.php (hunk 1 above)
Comment (felixheidecke) [reply apps/systemtags/appinfo/app.php]: It was already there

File: lib/private/legacy/template.php (hunk 2 above)
Comment (felixheidecke) [reply lib/private/legacy/template.php]: Done

**Review by PVince81 — APPROVED** [review]
:+1: thanks, please backport

**Pull request merged**